internal/node/reverse_proxy/restore_test.go

```diff
@@ -72,5 +72,5 @@ func TestProcessRestorer(t *testing.T) {
 	err = restorer.Restore(restoreUpdate)
 	assert.Nil(t, err)
 
-	assert.Equal(t, 1, len(restorer.ruleSet.rules))
+	assert.Equal(t, 2, len(restorer.ruleSet.rules))
```

Comment:
was this broken before?

Reply:
The behavior for restoring changed. Previously, it looked at started processes, and worked backwards to figure out what rules corresponded to them. The new behavior is to just look directly at the reverse proxy rules list directly. Imo, the reverse_proxy_rules state field should be the source of truth, and our transitions for stopping processes should be responsible for deactivating rules by removing them from the list.

The reason I made the change was because there are some default rules that don't correspond to processes which the old logic didn't handle.